web/devpi_web/description.py

```diff
@@ -28,11 +20,13 @@ def get_description(stage, name, version):
             html.a(link, href=link)).unicode(indent=2)
     metadata = stage.get_versiondata(name, version)
     desc = metadata.get("description")
-    if desc is None:
-        html = '<p>No description in metadata</p>'
-    else:
+    if desc:
         html = render_description(stage, desc)
-    return py.builtin._totext(html, "utf-8")
+    else:
+        html = '<p>No description in metadata</p>'
+    if not py.builtin._istext(html):
+        html = py.builtin._totext(html, "utf-8")
+    return html
```

Comment:
i btw think render_description should return unicode, and here we could not do the check (or just an assertion that it is unciode).

Reply:
I dislike asserts in production code, as they can be disabled with -o. With the check we don't care what we get back, we will always return unicode. I was just under the assumption that _totext would handle that already, but the _istext check is used in other parts of devpi as well.